app/javascript/mastodon/reducers/compose.js

```diff
@@ -106,7 +105,6 @@ function removeMedia(state, mediaId) {
 
   return state.withMutations(map => {
     map.update('media_attachments', list => list.filterNot(item => item.get('id') === mediaId));
-    map.update('text', text => text.replace(media.get('text_url'), '').trim());
```

Comment:
https://github.com/tootsuite/mastodon/pull/6672/files#diff-3fb459a28b27d6839fb62d18bceecf8aL104 This line is unnecessary.